src/test/groovy/graphql/execution/ValuesResolverTest.groovy

```diff
@@ -345,6 +346,162 @@ class ValuesResolverTest extends Specification {
+
+        where:
+        // from https://github.com/graphql/graphql-spec/pull/825/files#diff-30a69c5a5eded8e1aea52e53dad1181e6ec8f549ca2c50570b035153e2de1c43R1692
+        testCase                       | inputValue   | variables
+
+        '`{ a: null }` {}'             | buildObjectLiteral([
+                a: NullValue.of()
+        ])                                            | CoercedVariables.emptyVariables()
+
+        '`{ a: $var }`  { var : null}' | buildObjectLiteral([
+                a: VariableReference.of("var")
+        ])                                            | CoercedVariables.of(["var": null])
+
+    }
+
+    def "getArgumentValues: valid oneOf input - #testCase"() {
+        given: "schema defining input object"
+        def inputObjectType = newInputObject()
+                .name("oneOfInputObject")
+                .withAppliedDirective(Directives.OneOfDirective.toAppliedDirective())
+                .field(newInputObjectField()
+                        .name("a")
+                        .type(GraphQLString)
+                        .build())
+                .field(newInputObjectField()
+                        .name("b")
+                        .type(GraphQLInt)
+                        .build())
+                .build()
+        def fieldArgument = newArgument().name("arg").type(inputObjectType).build()
+
+        when:
+        def argument = new Argument("arg", inputValue)
+        def values = ValuesResolver.getArgumentValues([fieldArgument], [argument], variables, graphQLContext, locale)
+
+        then:
+        values == expectedValues
+
+        where:
+        // from https://github.com/graphql/graphql-spec/pull/825/files#diff-30a69c5a5eded8e1aea52e53dad1181e6ec8f549ca2c50570b035153e2de1c43R1692
+        testCase                       | inputValue   | variables                              | expectedValues
+
+        '{ b: 123 }` {}'               | buildObjectLiteral([
+                b: IntValue.of(123)
+        ])                                            | CoercedVariables.emptyVariables()      | [arg: [b: 123]]
+
+        '`$var` { var: { b: 123 } }'   | VariableReference.of("var")
+                                                      | CoercedVariables.of([var: [b: 123]])   | [arg: [b: 123]]
+
+        '{ a: "abc" }` {}'             | buildObjectLiteral([
+                a: StringValue.of("abc")
+        ])                                            | CoercedVariables.emptyVariables()      | [arg: [a: "abc"]]
+
+
+        '`$var` { var: { a: "abc" } }' | VariableReference.of("var")
+                                                      | CoercedVariables.of([var: [a: "abc"]]) | [arg: [a: "abc"]]
+
+        '{ a: $var }` { var : "abc"}'  | buildObjectLiteral([
+                a: VariableReference.of("var")
+        ])                                            | CoercedVariables.of([var: "abc"])      | [arg: [a: "abc"]]
+
+    }
+
```

Comment:
most of the tests are here - these are taken from the examples given in the spec update plus some they missed